src/main/java/com/epam/reportportal/service/LaunchLoggingContext.java

```diff
@@ -0,0 +1,155 @@
+package com.epam.reportportal.service;
+
+import com.epam.reportportal.message.TypeAwareByteSource;
+import com.epam.reportportal.restendpoint.http.MultiPartRequest;
+import com.epam.ta.reportportal.ws.model.BatchSaveOperatingRS;
+import com.epam.ta.reportportal.ws.model.Constants;
+import com.epam.ta.reportportal.ws.model.log.SaveLogRQ;
+import com.google.common.base.Strings;
+import com.google.common.net.MediaType;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+import org.reactivestreams.Publisher;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.epam.reportportal.utils.SubscriptionUtils.logFlowableResults;
+import static com.epam.reportportal.utils.files.ImageConverter.convert;
+import static com.epam.reportportal.utils.files.ImageConverter.isImage;
+import static com.google.common.io.ByteSource.wrap;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+public class LaunchLoggingContext {
+
+	/* default back-pressure buffer size */
+	public static final int DEFAULT_BUFFER_SIZE = 10;
+	public static final String DEFAULT_LAUNCH_KEY = "default";
+
+	static ConcurrentHashMap<String, LaunchLoggingContext> loggingContextMap = new ConcurrentHashMap<String, LaunchLoggingContext>();
```

Comment:
What is the reason for creating this map?

Reply:
It is possible to start a few launches in parallel. so we need to store context for them